src/migrations/20240420000000-correct-onApprovedAR-values.js

```diff
@@ -0,0 +1,331 @@
+const {
+  prepMigration,
+} = require('../lib/migration');
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await prepMigration(queryInterface, transaction, __filename);
+      await queryInterface.sequelize.query(/* sql */`
+
+      --  1. Calculate correct onApprovedAR values for objectives
+      DROP TABLE IF EXISTS objectives_on_ars;
+      CREATE TEMP TABLE objectives_on_ars
+      AS
+      SELECT
+        o.id oid,
+        BOOL_OR(ar.id IS NOT NULL AND ar."calculatedStatus" = 'approved') on_approved_ar,
+        BOOL_OR(ar.id IS NOT NULL) on_ar
+      FROM "Objectives" o
+      LEFT JOIN "ActivityReportObjectives" aro
+        ON o.id = aro."objectiveId"
+      LEFT JOIN "ActivityReports" ar
+        ON aro."activityReportId" = ar.id
+        AND ar."calculatedStatus" != 'deleted'
+      GROUP BY 1
+      ;
+      --  2. Calculate correct onApprovedAR values for goals
+      DROP TABLE IF EXISTS goals_on_ars;
+      CREATE TEMP TABLE goals_on_ars
+      AS
+      SELECT
+        g.id gid,
+        BOOL_OR(
+          (ar.id IS NOT NULL AND ar."calculatedStatus" = 'approved')
+          OR
+          COALESCE(ooaa.on_approved_ar,FALSE)
+        ) on_approved_ar,
+        BOOL_OR(ar.id IS NOT NULL OR COALESCE(ooaa.on_ar,FALSE)) on_ar
+      FROM "Goals" g
+      LEFT JOIN "ActivityReportGoals" arg
+        ON g.id = arg."goalId"
+      LEFT JOIN "ActivityReports" ar
+        ON arg."activityReportId" = ar.id
+        AND ar."calculatedStatus" != 'deleted'
+      LEFT JOIN "Objectives" o
+        ON o."goalId" = g.id
+      LEFT JOIN objectives_on_ars ooaa
+        ON ooaa.oid = o.id
+      GROUP BY 1
+      ;
+      --  3. Calculate onApprovedAR stats for objectives
```

Comment:
Seems you have alot of stats sql, I think this is a good idea. Just be careful that a sql error in your stats table/calc doesn't give you a false positive. Some times I find writing a new smaller query in the analyzer can help me re-think what I already have written.

Reply:
I'm not sure if I understand what you're saying, but I will note that:
- All the temp tables can be examined separately
- The calculations for the before/afters are performed in a different way from the calculations for the updates to lower the chance that they correlate spuriously